verifier/src/lib.rs

```diff
@@ -2,11 +2,334 @@
+    ///    fails the `right` will *still* be attempted.
+    /// * `right` - The right, or second, [`Verifier`] to perform.
+    pub fn new(left: L, right: R) -> Self {
+        Self { left, right }
+    }
+
+    /// The left side of this logical and instance.
+    pub fn left(&self) -> &L {
+        &self.left
+    }
+
+    /// The right side of this logical and instance.
+    pub fn right(&self) -> &R {
+        &self.right
+    }
+}
+
+impl<L: Verifier, R: Verifier> Verifier for And<L, R> {
+    type Error = AndError<L::Error, R::Error>;
+    fn verify(&self) -> CtOption<Self::Error> {
+        CtOption::new(
+            AndError::new(self.left.verify(), self.right.verify()),
+            1.into(),
+        )
+    }
+}
+
+/// An error that occurs during an `or` operation.
+///
+/// One should use the [`IntoResult::into_result()`] to determine if this is a
+/// failure or not.
+#[derive(Debug)]
+pub struct OrError<L, R> {
+    left: CtOption<L>,
+    right: CtOption<R>,
+}
+
+impl<L, R> OrError<L, R> {
+    /// Create a new instance
+    ///
+    /// Both the results of `left` and `right` can be `None`. Calling
+    /// [`IntoResult::into_result()`] on this instance will determine if this is
+    /// a failure or not.
+    pub fn new(left: CtOption<L>, right: CtOption<R>) -> Self {
+        Self { left, right }
+    }
+}
+
+impl<L, R> IntoResult<(), VerificationError> for OrError<L, R>
+where
+    L: IntoResult<(), VerificationError>,
+    R: IntoResult<(), VerificationError>,
+{
+    fn into_result(self) -> Result<(), VerificationError> {
+        let left = self.left.into_result();
+        let right = self.right.into_result();
+        match (left, right) {
+            (Err(_), Err(_)) => Err(VerificationError),
+            _ => Ok(()),
+        }
+    }
+}
+
+/// Will perform a logical or operation for the [`Verifier::verify()`]
+/// operation.
+///
+/// This is will be a long operation. If the `left` side succeeds
+/// the `right` side will *still* be exercised.
+#[derive(Debug)]
+pub struct Or<L, R> {
+    left: L,
+    right: R,
+}
+
+impl<L, R> Or<L, R> {
+    /// Create a new [`Or`] instance
+    ///
+    /// # Arguments:
+    /// * `left` - The left, or first, [`Verifier`] to perform. If this
+    ///    succeeds the `right` will *still* be attempted.
+    /// * `right` - The right, or second, [`Verifier`] to perform.
+    pub fn new(left: L, right: R) -> Self {
+        Self { left, right }
+    }
+
+    /// The left side of this logical or instance.
+    pub fn left(&self) -> &L {
+        &self.left
+    }
+
+    /// The right side of this logical or instance.
+    pub fn right(&self) -> &R {
+        &self.right
+    }
+}
+
+impl<L: Verifier, R: Verifier> Verifier for Or<L, R> {
+    type Error = OrError<L::Error, R::Error>;
+    fn verify(&self) -> CtOption<Self::Error> {
+        CtOption::new(
+            OrError::new(self.left.verify(), self.right.verify()),
+            1.into(),
+        )
```

Comment:
Same archetype as `And<L, R>`:
```suggestion
        let left_err = self.left.verify();
        let right_err = self.right.verify();
        let both_err = left_err.is_some() & right_err.is_some();
        CtOption::new(OrError::new(left_err, right_err), both_err)
```

Reply:
done, used `is_some` instead of `either_err` to match `CtOption::new()`